src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer<? extends Machine> machineProducer;
+        switch (type.getSimpleName()) {
+            case "Bulldozer":
+                machineProducer = new BulldozerProducer();
+                return new ArrayList<>(machineProducer.get());
+            case "Truck":
+                machineProducer = new TruckProducer();
+                return new ArrayList<>(machineProducer.get());
+            case "Excavator":
+                machineProducer = new ExcavatorProducer();
+                return new ArrayList<>(machineProducer.get());
+            default:
+                return new ArrayList<>();
+        }
```

Comment:
I'd recommend creating a map as a class field (using class as a key and producer as a value) and using this map instead of switch